docs(demo): add demo page (#466)
armory/docs: content/en/docs/Overview/demo/_index.md

aliases:
  - /spinnaker-install-admin-guides/architecture/
---

## Overview

Armory Enteprise for Spinnaker is an superset of Spinnaker, which itself is a collection of microservices working in concert to create a single continuous delivery / deployment service.

Installation and configuration of Armory Enterprise for Spinnaker can seem a daunting task for those not yet familiar with open-source Spinnaker, so we created Minnaker: a demo deployment of Spinnaker designed to run in a single instance (EC2 or other virtual machine).

## Features

What is included in Armory Easy Poc? 

Armory Enterprise 

- Infrastructure as Code (IaC) using gitops and Terraform
- Policy Driven Deployment (PDD) - for security and compliance (Security as Code)
- Declarative Spinnaker Application and Pipelines for (Pipelines as Code)
- Automatic EKS Provisioning through gitops
- Armory Agent for Kubernetes (delegate for kubernetes)
- Enterprise Armory Platform powered by Spinnaker
- Built in Kubernetes deployment namespace for testing (QA, Stage, Prod)
- Extension points through yaml (Jenkins, Git, JIRA, Docker hub)

> Minnaker uses [K3s](https://k3s.io/) as the kubernetes server. If you already have a kubernetes cluster for testing you can skip to [Install Minnaker](/docs/overview/demo/install-demo#install-minnaker).

Reviewer: ```suggestion
Minnaker uses [K3s](https://k3s.io/) as the Kubernetes server. If you already have a Kubernetes cluster for testing, you can skip to [Install Minnaker]({{< ref "install-demo#install-minnaker" >}}).
```

---
title: Self Service Demo with Armory Enterprise for Spinnaker
linkTitle: "Armory Demo"
no_list: true
description: "This demo is a fully featured proof of concept (PoC) of Armory Enterprise for Spinnaker™. Walk through installation, configuration, and deployment using sample applications."
aliases:
  - /spinnaker-install-admin-guides/architecture/
---

## Overview

> Armory Enteprise for Spinnaker is an superset of Spinnaker, which itself is a collection of microservices working in concert to create a single continuous delivery / deployment service.

Reviewer 2: ```suggestion
Armory Enteprise for Spinnaker is a superset of Spinnaker, which itself is a collection of microservices working in concert to create a single continuous delivery / deployment service.
```

\<nitpick /\>

Reviewer 3: Not a nitpick, a legitimate copy edit. +1

Installation and configuration of Armory Enterprise for Spinnaker can seem a daunting task for those not yet familiar with open-source Spinnaker, so we created Minnaker: a demo deployment of Spinnaker designed to run in a single instance (EC2 or other virtual machine).

## Features

> What is included in Armory Easy Poc? 

Reviewer 2: > Armory Easy Poc?

Should this be Minnaker?

Reviewer 3: Probably.

Armory Enterprise 

> - Infrastructure as Code (IaC) using gitops and Terraform

Reviewer 2: `leveraging` perhaps conveys better intent here. `gitops` isn't a tool so much as its a certain way of using IaC manifests in some kind of deployment

- Policy Driven Deployment (PDD) - for security and compliance (Security as Code)

> - Declarative Spinnaker Application and Pipelines for (Pipelines as Code)

Reviewer 2: Should Application be plural as well?

> - Automatic EKS Provisioning through gitops

Reviewer 2: What does this mean exactly, does the demo provision EKS clusters?

Reviewer 3: I think this whole section is more about what Spinnaker and/or Armory Enterprise can do, not what this Minnaker demo provides. What do you think, should it be edited to be specific to what this guide demonstrates, or to explicitly be about what we _can_ offer?

CC @armory/docs-team for comment.

> - Armory Agent for Kubernetes (delegate for kubernetes)

Reviewer 2: Should we link to agent documentation here?

> - Enterprise Armory Platform powered by Spinnaker

Reviewer 2: Could you help me understand what is meant with this feature?

- Built in Kubernetes deployment namespace for testing (QA, Stage, Prod)
- Extension points through yaml (Jenkins, Git, JIRA, Docker hub)

> Minnaker uses [K3s](https://k3s.io/) as the Kubernetes server. If you already have a Kubernetes cluster for testing, you can skip to [Install Minnaker]({{< ref "install-demo#install-minnaker" >}}).

Reviewer 2: ```suggestion
Minnaker uses [K3s](https://k3s.io/) as the Kubernetes implementation. If you already have a Kubernetes cluster for testing, you can skip to [Install Minnaker]({{< ref "install-demo#install-minnaker" >}}).
```

## {{% heading "prereq" %}}

Here are the requirements for completing this demo:

* You need to be able to provision instances in your cloud environment.
* The provisioned instance needs to meet the following minimum requirements
  * Ubuntu 18.04
  * 4 vCPUs
  * 16 GB of memory
  * 50 GB of storage

> If deploying to an EC2 instance, we suggest using the t2.xlarge tier.

Reviewer 2: If we're suggesting a default instance size can we make it `t3a.xlarge`? It is the newest and cheapest of the `t` offerings from Amazon, and [comparable in traditional specs](https://aws.amazon.com/ec2/pricing/on-demand/) to the `t2` class instances.

> If deploying to an EC2 instance, we suggest using the t2.xlarge tier.

Reviewer 2: ```suggestion
If deploying to an EC2 instance, we suggest using a t3a.xlarge instance.
```

> You don't need to create this instance ahead of following the guide, only confirm access to the resources required.

### Concepts

> The Minnaker demo uses the Armory Operator to manage services, and Kustomize to manage configuration files. If you're unfamiliar with either, check out the following docs before you being:

Reviewer 2: ```suggestion
The Minnaker demo uses the Armory Operator to manage services, and Kustomize to manage configuration files. If you're unfamiliar with either, check out the following docs before you begin:
```
